Add comments to SdaStorage.cs

hypixus opened the pull request:
Added spaces and other cool stuff.

TCJenkins commented:
Can one of the admins verify this patch?

Top-Cat commented:
Both your PRs are still missing the initial spaces after the ///

hypixus commented:
which ones do you mean? Please, give me an example, I can't find out bugs. i. e. Epple - e is where a should be - Apple

Top-Cat commented:
Like
- https://github.com/Top-Cat/UnofficialSteamAuthenticator/blob/13ef54b9f0aa7a6e7f1c8db3b771d28b2d65648b/SteamAuth/Models/User.cs#L5
- https://github.com/Top-Cat/UnofficialSteamAuthenticator/blob/13ef54b9f0aa7a6e7f1c8db3b771d28b2d65648b/SteamAuth/Models/Language.cs
- https://github.com/Top-Cat/UnofficialSteamAuthenticator/blob/13ef54b9f0aa7a6e7f1c8db3b771d28b2d65648b/SteamAuth/Properties/Annotations.cs#L53

hypixus commented:
@Top-Cat If you want, just copy a part of my project, in UsersPage.xaml I've made a region for this.

Top-Cat commented:
Newtonsoft should not be referenced in the main project, move the code that uses it into a helper in the library project, remove the while(true) and call the helper from the callback.

This will also not work with encrypted maFiles, there's already code to read them.

TCJenkins commented:
Can one of the admins verify this patch?

hypixus commented:
@Top-Cat Everything you asked for, point by point ;)
In void of button there is a TODO list before pulling changes to Windows Store.
P. S. Thank you for forcing me to use those callbacks, I finally understand how do they work :D

Top-Cat commented on UnofficialSteamAuthenticator/UsersPage.xaml.cs:
```diff
@@ -187,5 +191,84 @@ private void SettingsButton_Click(object sender, RoutedEventArgs e)
+            Even if it is empty, or couldn't be properly moved to json format - app
+            catches it and stops loading progress.
+             */
+            if (gotjson)
```
Nothing ever sets this, code should be in the callback below

Top-Cat commented on UnofficialSteamAuthenticator/packages.config:
```diff
@@ -3,4 +3,5 @@
   <package id="Microsoft.ApplicationInsights" version="1.0.0" targetFramework="wpa81" />
   <package id="Microsoft.ApplicationInsights.PersistenceChannel" version="1.0.0" targetFramework="wpa81" />
   <package id="Microsoft.ApplicationInsights.WindowsApps" version="1.0.0" targetFramework="wpa81" />
+  <package id="Newtonsoft.Json" version="8.0.3" targetFramework="wpa81" />
```
This is still here

Top-Cat commented on UnofficialSteamAuthenticator/UsersPage.xaml.cs:
```diff
@@ -187,5 +191,84 @@ private void SettingsButton_Click(object sender, RoutedEventArgs e)
+            2. Add resources with translated texts.
+            */
+        }
+        public void getLoadStatus(int status)
```
Return exception or null is probably more useful

Top-Cat commented on SteamAuth/SdaStorage.cs:
```diff
@@ -148,5 +150,70 @@ public sealed class SdaStorage
+                // if no exception caught - return success code.
+                callback(1);
+            }
+            catch (Exception)
```
I'm not a huge fan of pokemon

hypixus replied:
What do you mean here???

Top-Cat commented on UnofficialSteamAuthenticator/UsersPage.xaml.cs:
```diff
@@ -187,5 +191,84 @@ private void SettingsButton_Click(object sender, RoutedEventArgs e)
+        {
+            MessageDialog msgbox = new MessageDialog("Pick manifest file with account files.");
+            FileOpenPicker openPicker = new FileOpenPicker();
+            AddFileCallback filecallback = new AddFileCallback(getLoadStatus);
```
You should be able to just pass the function directly instead of wrapping it

Top-Cat commented on UnofficialSteamAuthenticator/UsersPage.xaml:
```diff
@@ -68,7 +68,8 @@
                               Icon="Add"
-                              Label="Add" />
+                              Label="Add new" />
+                <AppBarButton x:Name="appBarButton" Icon="Download" Label="Load file" Click="loadButton_Click"/>
```
Is this added to the en_US strings?

hypixus commented:
@Top-Cat 
1. Please don't expect me to add to language strings, I don't know how do they work and what to do with them.
2. What was the "pokemon" problem???

Top-Cat commented:
1. Just add it to the en_US xml, ignore the rest
2. Gotta catch 'em all (or don't, catch the specific exceptions)

hypixus commented:
@Top-Cat kkkkkkk Ok then ill repair it today. Add it to xml, and when to load the specific text?

hypixus commented:
@Top-Cat Everything done. Your problem with catch them all is handled by creating my own type of exception, called AddFileException. Also, I've removed not needed callbacks and added entries in en-US/Resources.resw.
